app/Http/Controllers/Service/Auth/ForgotPasswordController.php

```diff
@@ -37,4 +39,27 @@ public function showLinkRequestForm()
     {
         return view('service.auth.passwords.email');
     }
+
+    /**
+     * Override to make sure we don't back() by referrer
+     */
+    public function sendResetLinkEmail(Request $request)
+    {
+        $this->validate($request, ['email' => 'required|email']);
+
+        $response = $this->broker()->sendResetLink(
+            $request->only('email')
+        );
+
+        $back_url = $request->session()->get('_previous')['url'];
```

Comment:
Is this prone to similar security issue as back? Could that value be injected - or not easily because it is part of session?

Reply:
It appears to be the solution we use in MOMO, i think?